sensorimotor/sensorimotor/orphan_temporal_memory.py

```diff
@@ -0,0 +1,423 @@
+    """
+    activeCells = set()
+    winnerCells = set()
+
+    (_activeCells,
+     _winnerCells,
+     predictedColumns,
+     orphanCells) = self.activateCorrectlyPredictiveCells(
+       prevPredictiveCells,
+       prevMatchingCells,
+       activeColumns)
+
+    activeCells.update(_activeCells)
+    winnerCells.update(_winnerCells)
+
+    (_activeCells,
+     _winnerCells,
+     learningSegments,
+     chosenCellForColumn) = self.burstColumns(
+       activeColumns,
+       predictedColumns,
+       prevActiveCells | prevActiveExternalCells,
+       prevWinnerCells,
+       learnOnOneCell,
+       chosenCellForColumn,
+       connections)
+
+    activeCells.update(_activeCells)
+    winnerCells.update(_winnerCells)
+
+    if learn:
+      prevCellActivity = prevActiveExternalCells
+
+      if formInternalConnections:
+        prevCellActivity.update(prevWinnerCells)
+
+      self.learnOnSegments(prevActiveSegments,
+                           learningSegments,
+                           prevActiveCells | prevActiveExternalCells,
+                           winnerCells,
+                           prevCellActivity,
+                           connections,
+                           orphanCells,
+                           prevMatchingSegments)
+
+    (activeSegments,
+     predictiveCells,
+     matchingSegments,
+     matchingCells) = self.computePredictiveCells(
+       activeCells | activeExternalCells, connections)
+
+    return (activeCells,
+            winnerCells,
+            activeSegments,
+            predictiveCells,
+            predictedColumns,
+            chosenCellForColumn,
+            matchingSegments,
+            matchingCells)
+
+
+  def activateCorrectlyPredictiveCells(self,
+                                       prevPredictiveCells,
+                                       prevMatchingCells,
+                                       activeColumns):
+    """
+    Phase 1: Activate the correctly predictive cells.
+
+    Pseudocode:
+
+      - for each prev predictive cell
+        - if in active column
+          - mark it as active
+          - mark it as winner cell
+          - mark column as predicted
+        - if not in active column
+          - mark it as an orphan cell
+
+    @param prevPredictiveCells (set) Indices of predictive cells in `t-1`
+    @param activeColumns       (set) Indices of active columns in `t`
+
+    @return (tuple) Contains:
+                      `activeCells`      (set),
+                      `winnerCells`      (set),
+                      `predictedColumns` (set),
+                      `orphanCells`      (set)
+    """
+    activeCells = set()
+    winnerCells = set()
+    predictedColumns = set()
+    orphanCells = set()
+
+    for cell in prevPredictiveCells:
+      column = self.columnForCell(cell)
+
+      if column in activeColumns:
+        activeCells.add(cell)
+        winnerCells.add(cell)
+        predictedColumns.add(column)
+      elif cell in prevMatchingCells:
```

Comment:
Is this necessary given https://github.com/numenta/nupic.research/pull/148/files#diff-1968c3cead7c67353b112847117974e5R253?